lab-kyle/entry.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+// require webpack assets
+require('!!file?name=[name].[ext]!./html/index.html');
+require('./scss/base.scss');
+
+// npm modules
+const angular = require('angular');
+// app modules
+const cowsay = require('cowsay-browser');
+// module constants
+
+var app = angular.module('cowsayApp', []);
+// app logic
+
+app.controller('CowsayController', [ '$scope', function($scope){
+  $scope.cowsayText = 'angular rocks';
+  $scope.cowsayFile = 'default';
+  $scope.say = function(){
+    let text = $scope.cowsayText || 'Please fill out';
+    return cowsay.say({text: text, f: $scope.cowsayFile});
+  };
+
+  $scope.done = function(){
+    let text = $scope.cowsayText || 'Fill this out too.';
+    $scope.result = cowsay.say({text: text , f: $scope.cowsayFile});
```

Comment:
You can DRY this out a bit. What you have here in $scope.done() is essentially exactly the same as what you have in $scope.say(). So I would recommend to change it to:

$scope.done = function() {
  $scope.result = $scope.say();
};